O365OrgSettings: Empty InstallationOptionsAppsForMac & InstallationOptionsAppsForWindows failing

### Description of the issue

If I try to disable the settings InstallationOptionsAppsForMac and InstallationOptionsAppsForWindows using "@()", the SET operation fails. This error has been occurring since version 1.26.114.1.

### Microsoft 365 DSC Version

V1.26.114.1

### Which workloads are affected

Office 365 Admin

### The DSC configuration

```powershell
@{
    NonNodeData = @{
        Office365 = @{
            OrgSettings = @{

                InstallationOptionsUpdateChannel                = 'monthlyEnterprise'
                InstallationOptionsAppsForWindows               = @()
                InstallationOptionsAppsForMac                   = @()

            }

        }
    }
}
```

### Verbose logs showing the problem

```powershell
##[error]Cannot bind argument to parameter 'ReferenceObject' because it is null.
    + CategoryInfo          : InvalidData: (:) [], CimException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.CompareObje 
   ctCommand
    + PSComputerName        : localhost

##[error]The PowerShell DSC resource '[O365OrgSettings]OrgSettingsDefaults::[Office365]Office365_Configuration' with SourceInfo 
'::3::1::O365OrgSettings' threw one or more non-terminating errors while running the Set-TargetResource functionality. 
These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more 
details
```

### Environment Information + PowerShell Version

```powershell
OsName               : Microsoft Windows Server 2022 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsVersion       : 2009
WindowsBuildLabEx    : 20348.1.amd64fre.fe_release.210507-1500
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

Name                           Value
----                           -----
PSVersion                      5.1.20348.4294
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.20348.4294
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Change log for Microsoft365DSC
 
+# UNRELEASED
+
+* O365OrgSettings
+  * Fixed an issue where comparing empty app installation options failed.
+    FIXES [#6812](https://github.com/microsoft/Microsoft365DSC/issues/6812)
+
 # 1.26.121.1
 
 * AADGroup

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_O365OrgSettings/MSFT_O365OrgSettings.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_O365OrgSettings/MSFT_O365OrgSettings.psm1
@@ -303,8 +303,8 @@ function Get-TargetResource
 
             $results += @{
                 InstallationOptionsUpdateChannel  = $installationOptions.updateChannel
-                InstallationOptionsAppsForWindows = $appsForWindowsValue | Sort-Object
-                InstallationOptionsAppsForMac     = $appsForMacValue | Sort-Object
+                InstallationOptionsAppsForWindows = @($appsForWindowsValue | Sort-Object)
+                InstallationOptionsAppsForMac     = @($appsForMacValue | Sort-Object)
             }
         }
 

